src/module/normalisation/heap_profiler.rs

```diff
@@ -0,0 +1,219 @@
+use log::error;
+use std::collections::HashMap;
+use walrus::*;
+
+/// Modifies the WASM binary to add heap profiling support. Every time one of allocation function
+/// is called:
+///  * malloc(arg1) -> ret
+///  * aligned_alloc(arg1, arg2) -> ret
+///  * calloc(arg1, arg2) -> ret
+///  * realloc(arg1, arg2) -> ret
+///  * free(arg1)
+/// extra call to its profiling function will be invoked:
+///  * malloc_profiler(arg1, ret)
+///  * aligned_alloc_profiler(arg1, arg2, ret)
+///  * calloc_profiler(arg1, arg2, ret)
+///  * realloc_profiler(arg1, arg2, ret)
+///  * free_profiler(arg1)
+///
+/// Profiling functions are imported from heap_profiler module.
+///
+/// Functions that can't be found in a module are ignored and error is logged.
+pub fn patch(module: &mut Module) {
+    // NOTE rusts global allocator __rust_alloc sometimes will
+    // execute aligned_alloc instead of malloc
+    ["malloc", "aligned_alloc", "calloc", "realloc", "free"]
+        .iter()
+        .for_each(|name| {
+            add_profiler_to(module, name).unwrap_or_else(|e| error!("{}", e));
```

Comment:
currently normalisation patch will log error instead of fail not to break tests. Better solution would be to let patch fail and refactor normalisation tests instead.

As profiler is not critical part of the system atm, simpler solution was chosen